internal/controller/pipelinerollout_controller.go

```diff
@@ -732,20 +731,24 @@ func (r *PipelineRolloutReconciler) updatePipelineRolloutStatusToFailed(ctx cont
 }
 
 // getPipelineName retrieves the name of the current running pipeline managed by the given
-// pipelineRollout through the `promoted` label. If no such pipeline exists, then it
-// constructs the name by calculating the suffix and appending to the PipelineRollout name.
-func (r *PipelineRolloutReconciler) getPipelineName(ctx context.Context, pipelineRollout *apiv1.PipelineRollout) (string, error) {
+// pipelineRollout through the `promoted` label. Unless there is non such pipeline exist, then
+// construct the name by calculate the suffix and append to the PipelineRollout name.
+func (r *PipelineRolloutReconciler) getPipelineName(
+	ctx context.Context,
+	pipelineRollout *apiv1.PipelineRollout,
+	upgradeState string,
+) (string, error) {
```

Comment:
This naming seems good. So, if I'm following along correctly:
1. we create our very first Pipeline for the PipelineRollout as "<pipeline>-1" with Label "promoted". 
2. Then say we make a Direct apply change: we simply update "<pipeline>-1" since it is the one with Label "promoted".
3. Then say we need to make a Progressive change: we don't find a Pipeline labeled "upgrading", so we create a new one "<pipeline>-2" with label "upgrading".
4. If for some reason somebody makes a change to the Pipeline spec before "<pipeline>-2" is "Running" or "Failed", then we update that spec again. 
5. Once "<pipeline>-2" is "Running", we mark it "promoted". 
6. Next time we need a Progressive update, we notice that there's no "upgrading" state so we create a new Pipeline "<pipeline>-3"

Reply:
yes, that's correct.